src/ragger/backend/interface.py

```diff
@@ -359,3 +383,28 @@ def navigate_until_snap(self,
         :rtype: NoneType
         """
         raise NotImplementedError
+
+    @abstractmethod
+    def navigate_and_compare(self, path: Path, test_case_name: Path,
+                             instructions: List[NavigationInstruction]) -> bool:
+        """
+        Navigate and compare snapshots.
```

Comment:
Make it explicit that there are as many navigation steps as provided by the instructions parameter